libs/block-mesh-manager/src/worker/analytics_agg.rs

```diff
@@ -17,19 +18,22 @@ pub async fn analytics_agg(
     pool: PgPool,
     mut rx: tokio::sync::mpsc::Receiver<AnalyticsMessage>,
 ) -> Result<(), anyhow::Error> {
-    let mut queries: Vec<String> = Vec::with_capacity(100);
-    let mut calls: HashMap<Uuid, AnalyticsMessage> = HashMap::new();
-    let mut count = 0;
     let agg_size = env::var("AGG_SIZE")
         .unwrap_or("100".to_string())
         .parse()
         .unwrap_or(100);
-
+    let mut queries: Vec<String> = Vec::with_capacity(agg_size * 3);
+    let mut calls: HashMap<Uuid, AnalyticsMessage> = HashMap::new();
+    let mut count = 0;
+    let mut prev = Utc::now();
     while let Some(query) = rx.recv().await {
-        tracing::info!("HELLO {:#?}", query);
         calls.insert(query.user_id, query.clone());
         count += 1;
-        if count == agg_size {
+        let now = Utc::now();
+        let diff = now - prev;
+        let run = diff.num_seconds() > 5 || count >= agg_size;
+        prev = Utc::now();
+        if run {
```

Comment:
adding time limit too incase not many requests are inbound

only caveat is that it still needs a new message to trigger it